src/web/app/components/question-types/question-statistics/mcq-question-statistics.component.spec.ts

```diff
@@ -24,4 +25,62 @@ describe('McqQuestionStatisticsComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should calculate statistics correctly', () => {
+    component.question.mcqChoices = ['optionA', 'optionB', 'optionC'];
+    component.question.otherEnabled = false;
+    component.question.hasAssignedWeights = true;
+    component.question.mcqWeights = [1, 2, 3];
+    component.responses = JSON.parse(JSON.stringify(responses.responsesNoOther));
```

Comment:
is it necessary to `JSON.stringify` here?

Reply:
not using JSON.stringify results in an error to occur when running tests (as shown below), hence I left the JSON.stringify here. I'm not too familiar with JSON files, so I was unsure if there was a cleaner way to parse the data. Do seek your input if there is!

![image](https://user-images.githubusercontent.com/65604398/127732693-22c767a5-4373-4377-9559-16bafd2d23b7.png)

Reply:
@Jellybeano it seems like `responses.responsesNoOther` is already a javascript object, can you try using `component.responses = responses.responsesNoOther;` ?

Reply:
@daongochieu2810 because the json file contains lines such as `"questionType": "FeedbackQuestionType.RUBRIC"`, the following error will appear. That is why I initially chose to use json parse and json stringify within the code

![image](https://user-images.githubusercontent.com/65604398/129475584-5e50cdc0-9cd1-4660-bdd6-c3b6115e4ca7.png)

Reply:
@Jellybeano How about this `component.responses = responses.responsesNoOther as FeedbackQuestionType;`

Reply:
@daongochieu2810 thanks for the suggestion! managed to get it to work now